packages/framework/src/Controller/Admin/DeliveryAddressController.php

```diff
@@ -5,21 +5,85 @@
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
 class DeliveryAddressController extends AdminBaseController
 {
     /**
      * @param \Shopsys\FrameworkBundle\Model\Customer\DeliveryAddressFacade $deliveryAddressFacade
+     * @param \Shopsys\FrameworkBundle\Model\Customer\DeliveryAddressDataFactory $deliveryAddressDataFactory
+     * @param \Shopsys\FrameworkBundle\Model\AdminNavigation\BreadcrumbOverrider $breadcrumbOverrider
+     * @param \Shopsys\FrameworkBundle\Model\Customer\CustomerFacade $customerFacade
      */
     public function __construct(
         protected readonly DeliveryAddressFacade $deliveryAddressFacade,
+        protected readonly DeliveryAddressDataFactory $deliveryAddressDataFactory,
+        protected readonly BreadcrumbOverrider $breadcrumbOverrider,
+        protected readonly CustomerFacade $customerFacade,
     ) {
     }
 
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param int $id
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    #[Route(path: '/delivery-address/edit/{id}', requirements: ['id' => '\d+'], name: 'admin_delivery_address_edit')]
+    public function editAction(Request $request, int $id): Response
+    {
+        $deliveryAddress = $this->deliveryAddressFacade->getById($id);
+        $deliveryAddressData = $this->deliveryAddressDataFactory->createFromDeliveryAddress($deliveryAddress);
+
+        $form = $this->createForm(DeliveryAddressFormType::class, $deliveryAddressData, [
+            'domain_id' => $deliveryAddress->getCustomer()->getDomainId(),
+        ]);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->deliveryAddressFacade->edit($id, $deliveryAddressData);
+
+            $this->addSuccessFlashTwig(
+                t('Customer <strong><a href="{{ url }}">{{ name }}</a></strong> modified'),
+                [
+                    'name' => $deliveryAddress->getCity(),
+                    'url' => $this->generateUrl('admin_delivery_address_edit', ['id' => $deliveryAddress->getId()]),
+                ],
+            );
+
+            $customer = $deliveryAddress->getCustomer();
+
+            if ($this->customerFacade->isB2bFeaturesEnabledByCustomer($customer)) {
+                $billingAddress = $customer->getBillingAddress();
+
+                return $this->redirectToRoute('admin_billing_address_edit', ['id' => $billingAddress->getId()]);
+            }
+
+            return $this->redirectToRoute('admin_customer_list');
+        }
+
+        if ($form->isSubmitted() && !$form->isValid()) {
+            $this->addErrorFlashTwig(t('Please check the correctness of all data filled.'));
+        }
+
+        $this->breadcrumbOverrider->overrideLastItem(
+            t('Editing customer - %name%', ['%name%' => $deliveryAddress->getCity()]),
```

Comment:
`editing delivery address - ...` maybe?

Reply:
:+1: fixed in https://github.com/shopsys/shopsys/pull/3261/commits/551ee214613678f97967fd005045005e995e4f57